repository/zookeeper/operator/templates/bootstrap.sh.yaml

```diff
@@ -0,0 +1,322 @@
+        --data_log_dir      The directory where the ZooKeeper process will store its
+                            write ahead log. The default is
+                            /var/lib/zookeeper/data/log.
+        --conf_dir          The directoyr where the ZooKeeper process will store its
+                            configuration. The default is /opt/zookeeper/conf.
+        --client_port       The port on which the ZooKeeper process will listen for
+                            client requests. The default is 2181.
+        --election_port     The port on which the ZooKeeper process will perform
+                            leader election. The default is 3888.
+        --server_port       The port on which the ZooKeeper process will listen for
+                            requests from other servers in the ensemble. The
+                            default is 2888.
+        --tick_time         The length of a ZooKeeper tick in ms. The default is
+                            2000.
+        --init_limit        The number of Ticks that an ensemble member is allowed
+                            to perform leader election. The default is 10.
+        --sync_limit        The maximum session timeout that the ensemble will
+                            allows a client to request. The default is 5.
+        --heap              The maximum amount of heap to use. The format is the
+                            same as that used for the Xmx and Xms parameters to the
+                            JVM. e.g. --heap=2G. The default is 2G.
+        --max_client_cnxns  The maximum number of client connections that the
+                            ZooKeeper process will accept simultaneously. The
+                            default is 60.
+        --snap_retain_count The maximum number of snapshots the ZooKeeper process
+                            will retain if purge_interval is greater than 0. The
+                            default is 3.
+        --purge_interval    The number of hours the ZooKeeper process will wait
+                            between purging its old snapshots. If set to 0 old
+                            snapshots will never be purged. The default is 0.
+        --max_session_timeout The maximum time in milliseconds for a client session
+                            timeout. The default value is 2 * tick time.
+        --min_session_timeout The minimum time in milliseconds for a client session
+                            timeout. The default value is 20 * tick time.
+        --log_level         The log level for the zookeeeper server. Either FATAL,
+                            ERROR, WARN, INFO, DEBUG. The default is INFO.
+    "
+    }
+
+    function create_data_dirs() {
+        if [ ! -d $DATA_DIR  ]; then
+            mkdir -p $DATA_DIR
+            chown -R $USER:$USER $DATA_DIR
+        fi
+
+        if [ ! -d $DATA_LOG_DIR  ]; then
+            mkdir -p $DATA_LOG_DIR
+            chown -R $USER:$USER $DATA_LOG_DIR
```

Comment:
seems like a good place to refactor for all the dirs that you're a doing `mkdir` and `chown`?